src/azure-cli/azure/cli/command_modules/storage/oauth_token_util.py

```diff
@@ -1,48 +0,0 @@
-# --------------------------------------------------------------------------------------------
-# Copyright (c) Microsoft Corporation. All rights reserved.
-# Licensed under the MIT License. See License.txt in the project root for license information.
-# --------------------------------------------------------------------------------------------
-
-import threading
-
-
-class TokenUpdater(object):
```

Comment:
I am actually curios about the design of this `TokenUpdater`. Why does it actively refresh the token instead of passively 
 refreshing it on demand? Be careful of pitfalls that may be introduced by this change.

Reply:
refreshing it when expired. Here the code is changed to keep previous token credential for track 1 SDK.